modules/internal/String.chpl

```diff
@@ -875,6 +886,57 @@ module String {
       return this:c_string; // folded out in resolution
     }
 
+    /*
+      Returns a :record:`~Bytes.bytes` from the given :record:`string`. If the
+      string contains some escaped non-UTF8 bytes, `errors` argument determines
+      the action.
+        
+      :arg errors: `encodePolicy.pass` directly copies the data,
+                   `encodePolicy.unescape` recovers the escaped bytes back.
+
+      :returns: :record:`~Bytes.bytes`
+    */
+    proc encode(errors=encodePolicy.pass): bytes {
+      var localThis: string = this.localize();
+
+      if errors == encodePolicy.pass {  // just copy
+        return createBytesWithNewBuffer(localThis.buff, localThis.numBytes);
+      }
+      else {  // see if there is escaped data in the string
+        var (buf, size) = bufferAlloc(this.len+1);
+
+        var readIdx = 0;
+        var writeIdx = 0;
+        while readIdx < localThis.len {
+          var cp: int(32);
+          var nbytes: c_int;
+          var multibytes = (localThis.buff + readIdx): c_string;
+          var maxbytes = (localThis.len - readIdx): ssize_t;
+          // TODO make the following a separate function
+          const decodeRet = qio_decode_char_buf_esc(cp, nbytes, multibytes,
+                                                    maxbytes);
+          if (cp>=0xdc80 && cp<=0xdcff) {
```

Comment:
I always write such comparisons like `a < b && b < c`, because it's closer to normal math: `if (0xdc80<=cp && cp<=0xdcff)`